JEasyCryptoConsole/src/EasyCryptoConsole/EasyCryptoConsole.java

```diff
@@ -6,58 +6,59 @@
-
-			while (true) {
-				String d = console.readLine("Do you wish to encrypt or decrypt (e or d)? > ");
-				
-
-				EasyCryptoAPI.Result result;
-
-				if (d.equalsIgnoreCase("e")) {
-					String e = console.readLine("Please enter text to be encrypted > ");
-					String m = console.readLine("Please enter encryption method > ");
-					result = EasyCryptoAPI.encrypt(e, m);
-				} else if (d.equalsIgnoreCase("d")) {
-					String e = console.readLine("Please enter text to be decrypted > ");
-					String m = console.readLine("Please enter decryption method > ");
-					result = EasyCryptoAPI.decrypt(e, m);
-				} else {
-					return;
-				}
-		        
-				console.printf("\nResult is: %d - %s\n", result.resultCode().ordinal(), result.resultCode().toString());
-				switch (result.resultCode()) {
-				case ESuccess: {
-					console.printf("Encrypted text is: %s \n", result.result());
-					System.out.println(result.result());
-					break;
-				}
-				case EError:
-				case ENotSupported: {
-					console.printf("Encrypted text is: %s \n", result.result());
-					break;
-				}
-				}
-			}
-
-		} catch (UnsupportedEncodingException e1) {
-			e1.printStackTrace();
-		} catch (Exception e) {
-			e.printStackTrace();
-		}
-	}
+                    if (d.equalsIgnoreCase("e") || d.equalsIgnoreCase("d") || d.equalsIgnoreCase("q")) {
+                        if (d.equalsIgnoreCase("e")) {
+                            String e = console.readLine("Please enter text to be encrypted > ");
+                            String m = console.readLine("Please enter encryption method > ");
+                            result = EasyCryptoAPI.encrypt(e, m);
+                        } else if (d.equalsIgnoreCase("d")) {
+                            String e = console.readLine("Please enter text to be decrypted > ");
+                            String m = console.readLine("Please enter decryption method > ");
+                            result = EasyCryptoAPI.decrypt(e, m);
+                        } else {
+                            return;
+                        }
 
+                        console.printf("\nResult is: %d - %s\n", result.resultCode().ordinal(), result.resultCode().toString());
+                        switch (result.resultCode()) {
+                            case ESuccess: {
+                                console.printf("Encrypted text is: %s \n", result.result());
+                                System.out.println(result.result());
+                                break;
+                            }
+                            case EError:
+                            case ENotSupported: {
+                                console.printf("Encrypted text is: %s \n", result.result());
+                                break;
+                            }
+                        }
+                    } else {
+                        console.printf("Wrong input! Please type in 'e' or 'd' or 'q'!\n");
```

Comment:
This seems to be an error. We should use error print for that.

Reply:
Can you be more specific? What do you mean with error print? In this case user is not yet trying to encrypt or decrypt text so it seems logical that the input is handled separately and user is given feedback when input does not match any command.

Reply:
In Java we have `PrintSteams` for `System.out` and `System.err`. Even though both are printed to the console, it is from a debugging standpoint beneficial if we separate those (in case of error messages).

For more info see the [documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/System.html#err) and a relevant SO [post](https://stackoverflow.com/questions/21197737/what-is-the-point-of-system-err).